### PR TITLE
Add an autostart object

### LINKED ISSUE #762: Implement an autostart object
This `autostart` object should be responsible for starting the autostart and it should also:

* allow to see and change the autostart path (`autostart.path`)
* keep track whether the actual autostart process is still running
* show this via an attribute (`autostart.running`, maybe `autostart.pid`)
* if `autostart.auto_lock` is set (which I'd set per default), the autostart object takes care of `lock`ing/`unlock`ing the monitor state depending on whether the autostart is running
* deny executing `reload` while  the autostart is still running

### PR DESCRIPTION
**Pull request by t-wissmann**
This adds an autostart object which allows modifying the custom
autostart path at run time and also helps querying information about the
last invoked autostart run (pid, status, running).

This closes #762.

### CODE REVIEW

**Comment by t-wissmann** [comment]
Maybe `last_status` can be simplified to `status` or `exit_status`.

File: src/autostart.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    , last_status_(this, "last_status", 0)
+{
+    autostart_path_.setWritable();
+    autostart_path_.setDoc(
+                "Custom path to the user\'s autostart path. "
+                "If it is empty, then the autostart in $XDG_CONFIG_HOME "
+                "or $HOME is used.");
+    global_autostart_path_.setDoc("Path of the system-wide autostart, used as a fallback.");
+    pid_.setDoc(
+                "the process id of the last autostart invokation. "
```
Comment (blarz) [inline src/autostart.cpp:24]: invocation

File: src/main.cpp [hunk 2]
```diff
@@ -464,8 +420,9 @@ int main(int argc, char* argv[]) {
     if (g.trueTransparency) {
         X->tryInitTransparency();
     }
-    // remove zombies on SIGCHLD
-    sigaction_signal(SIGCHLD, remove_zombies);
+    // ignore sigchld, but set it anyway such that we don't
+    // receive anything on on stopped children (SA_NOCLDSTOP)
```
Comment (blarz) [inline src/main.cpp:424]: on on

**Review by blarz — CHANGES_REQUESTED** [review]
Some small typos, otherwise looks good!

Also: there are some autostart related tests in test_herbstluftwm.py, maybe they can be removed or update?

**Comment by blarz** [comment]
> Some small typos, otherwise looks good!
> 
> Also: there are some autostart related tests in test_herbstluftwm.py, maybe they can be removed or update?

At least the `# TODO: verify the path as soon as we have an autostart object` comment seems like something that could be done (maybe in a different PR?)

**Comment by t-wissmann** [comment]
I've fixed the typos and moved the test cases to `test_autostart.py`. The comment in `test_autostart_path` however requires another attribute that does not exist yet, so I think this should be done in separate PR.

**Comment by blarz** [comment]
> I've fixed the typos and moved the test cases to `test_autostart.py`. The comment in `test_autostart_path` however requires another attribute that does not exist yet, so I think this should be done in separate PR.

Thanks for the fixes. And makes sense :)

**Review by blarz — APPROVED** [review]

**Pull request merged**